File: runme/mcgen/local/datan/datsrc/lsqexp.c
#include <math.h>
#include "datsrc.h"

double LIBFUNC lsqexp_(double const P_T *x, integer const P_T *nr,
               double const P_T *t)
{
    /* System generated locals */
    double ret_val;

    /* Local variables */
    double arg;

    /* Parameter adjustments */
    --x;

    /* Function Body */
    arg = x[2] * *t;
    if (arg > 700.) {
        ret_val = 0.;
    } else {
        ret_val = x[1] * exp(-arg);
    }
    return ret_val;
} /* lsqexp_ */

File: runme/mcgen/local/datan/datsrc/lsqexp.f
      DOUBLE PRECISION FUNCTION LSQEXP(X,NR,T)
      IMPLICIT DOUBLE PRECISION (A-H,O-Z)
      DIMENSION X(NR)
      PARAMETER (BIG=700.D0,ZERO=0.D0)
      ARG=X(2)*T
      IF(ARG.GT.BIG) THEN
        LSQEXP=ZERO
      ELSE
        LSQEXP=X(1)*EXP(-ARG)
      END IF
      END
